src/Search.php

```diff
@@ -5165,7 +5165,7 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
             case "glpi_ticketvalidations.status":
             case "glpi_changes.global_validation":
             case "glpi_changevalidations.status":
-                if ($val == 'all') {
+                if ($val == 'all' || !is_numeric($val)) {
```

Comment:
There is a `can` case in the following switch.
```suggestion
                if (!in_array($val, ['all', 'can'], true) || !is_numeric($val)) {
```

Reply:
```suggestion
                if (!is_numeric($val)) {
```